### PR TITLE
switches from React.createClass to extend React.Component

### PR DESCRIPTION
**Pull request by Jascha-Sundaresan**
React.createClass is deprecated.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -13,57 +13,62 @@ function factory(options) {
...
-          () => proxyElements[this.props.proxy] || element,
-          this.props.timeline,
-          this.props);
-      } else {
-        spark(element, () => element, this.props.timeline, this.props);
+  const sparkScrollFactory = (defaultComponent) => {
+    return class SparkScroll extends React.Component {
+      constructor(props) {
+        super(props);
+      }
```
Comment (gilbox) [inline src/index.js]: the constructor is unecessary

File: src/index.js [hunk 2]
```diff
@@ -13,57 +13,62 @@ function factory(options) {
...
   const SparkScroll = sparkScrollFactory('div');
   SparkScroll.div = SparkScroll;
 
-  const sparkProxyFactory = defaultComponent => React.createClass({
-    displayName: 'SparkProxy.' + (_isString(defaultComponent) ? defaultComponent : defaultComponent.displayName),
+  const sparkProxyFactory = (defaultComponent) => {
+    return class SparkProxy extends React.Component {
+      constructor(props) {
+        super(props);
+      }
```
Comment (gilbox) [inline src/index.js]: this is also unecesary

**Review by gilbox — CHANGES_REQUESTED** [review]
thanks! just two minor things

**Comment by Jascha-Sundaresan** [comment]
Ok.. I removed the unnecessary constructors.

I also updated everything in package.json except babel and webpack as I didn't want to figure out how to fix the breaking changes.

react 15 threw some warnings about tags getting unknown props, so I fixed that, too.

**Comment by Jascha-Sundaresan** [comment]
sweet!

**Comment by gilbox** [comment]
thanks for your help! I just published v4.0.0

**Comment by Jascha-Sundaresan** [comment]
awesome.  My pleasure.

**Pull request merged**